### PR TITLE
CVSB-4867 & CVSB-7049

### PR DESCRIPTION
**Pull request by agoncearenco**
This service will write to a new SQS queue (update-status)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
   "husky": {
     "hooks": {
       "pre-commit": "npm run security-checks",
```
[the remark this answers is not in the document]
Comment (agoncearenco) [reply package.json]: You're right, doing it

[the remark this answers is not in the document]
Comment (agoncearenco) [reply package.json]: Requested change completed.

File: src/functions/certGenInit.ts [hunk 2]
```diff
@@ -18,15 +19,19 @@ const certGenInit: Handler = async (event: any, context?: Context, callback?: Ca
     }
 
     // Convert the received event into a readable array of filtered test results
-    const records: any[] = StreamService.getTestResultStream(event);
+    const expandedRecords: any[] = StreamService.getTestResultStream(event);
+    const certGenfilteredRecords: any[] = Utils.filterCertificateGenerationRecords(expandedRecords);
```
Comment (KHodderDVSA) [inline src/functions/certGenInit.ts]: certGenfilteredRecords should really be certGenFilteredRecords (capital F)

File: src/utils/Utils.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+     * Filter records to be sent to SQS
+     * @param records
+     */
+    public static filterCertificateGenerationRecords(records: any[]) {
+        return records
+            .filter((record: any) => { // Filter by testStatus
+                return record.testStatus === "submitted";
+            })
+            .filter((record: any) => { // Filter by testResult (abandoned tests are not allowed)
+                return (record.testTypes.testResult === "pass" || record.testTypes.testResult === "fail" || record.testTypes.testResult === "prs");
```
Comment (KHodderDVSA) [inline src/utils/Utils.ts]: 2 out of 3 conditions tested
image

File: src/functions/certGenInit.ts (hunk 2 above)
Comment (agoncearenco) [reply src/functions/certGenInit.ts]: Done

File: src/utils/Utils.ts (hunk 3 above)
Comment (agoncearenco) [reply src/utils/Utils.ts]: Done:
image

**Pull request merged**